bqskit/utils/math.py

```diff
@@ -166,3 +166,43 @@ def pauli_expansion(H: npt.NDArray[np.complex128]) -> npt.NDArray[np.float64]:
     A = np.stack(flatten_paulis, axis=-1)
     X = np.real(np.matmul(np.linalg.inv(A), flatten_H))
     return np.array(X)
+
+
+def kron(MatrixList):  # Kronecker product of list of matrices
```

Comment:
When the Kronecker product of two unitary matrices is needed, e.g XX, UnitaryMatrix.otimes works just fine. When there are more then two matrices, or the matrices are hermitian but not unitary, kron method should be used instead.  Note that for the case of multiple unitary matrices once can do repeated application of  UnitaryMatrix.otimes as well.

Reply:
`UnitaryMatrix.otimes` supports more than two matrices: `U.otimes(U1, U2, ...)`. You are right about the hermitian input, but it might be time to create a `HermitianMatrix`. However, I don't see any references to this function in your PR, am I missing something?

Reply:
Yes, I was not aware initially of the otimes functionality within UnitarityMatrix. In my PR I was using kron for that purpose and also thinking about two qudit rotations with the generators of SU(N) which are not always unitary, but always Hermitian (I'm using the physics convention). I did not comment about that in my PR though, you're right. I'm in favor of having a HermitianMatrix class for such cases. Also, in the case of SU(3), I have included the rotation by a single generator  as a parametrized gate. And, I plan to include two qudit rotations as well.

Reply:
If both UnitaryMatrix and HermitianMatrix are present and have kronecker products associated with them, I do not see any need for the kron function in math.py